Add apply CLI (#11376)
explosion/spaCy: website/docs/api/cli.md

in a `DocBin`. The input can be a single file or a directory. The recognized
input formats are:

1. `.spacy`
2. `.jsonl` containing a user specified `text_key`
3. Files with any other extension are assumed to be plain text files containing
   a single document.

When a directory is provided it is traversed recursively to collect all files.

```cli
$ python -m spacy apply [model] [data-path] [output-file] [--code] [--text-key] [--force-overwrite] [--gpu-id] [--batch-size] [--n-process]
```

| Name                                      | Description                                                                                                                                                                          |
| ----------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `model`                                   | Pipeline to apply to the data. Can be a package or a path to a data directory. ~~str (positional)~~                                                                                  |

> | `data_path`                               | Location of evaluation data in spaCy's [binary format](/api/data-formats#training). ~~Path (positional)~~                                                                            |

Reviewer: ```suggestion
| `data_path`                               | Location of data to be evaluated in spaCy's [binary format](/api/data-formats#training), jsonl, or plain text. ~~Path (positional)~~                                                                            |
```

| `output-file`, `-o`                       | Output `DocBin` path. ~~str (positional)~~                                                                                                                                           |
| `--code`, `-c` <Tag variant="new">3</Tag> | Path to Python file with additional code to be imported. Allows [registering custom functions](/usage/training#custom-functions) for new architectures. ~~Optional[Path] \(option)~~ |

> | `--text-key`, `-tk`                       | The key for `.jsonl` files to use to grab the texts from. ~~Optional[str] \(option)~~                                                                                                |

Reviewer: ```suggestion
| `--text-key`, `-tk`                       | The key for `.jsonl` files to use to grab the texts from. Defaults to `text`. ~~Optional[str] \(option)~~                                                                                                |
```

| `--force-overwrite`, `-F`                 | If the provided `output-file` already exists, then force `apply` to overwrite it. If this is `False` (default) then quits with a warning instead. ~~bool (flag)~~                    |

> | `--gpu-id`, `-g`                          | GPU to use, if any. Defaults to `-1` for CPU. ~~int (option)~~                                                                                                                       |

Reviewer 2: WIll need one more run of `prettier` ;-)

> | `--batch-size`, `-g`                      | Batch size to use for prediction. Defaults to `1`. ~~int (option)~~                                                                                                                  |
> | `--n-process`, `-g`                       | Number of processes to use for prediction. Defaults to `1`. ~~int (option)~~                                                                                                         |

Reviewer: ```suggestion
| `--batch-size`, `-b`                      | Batch size to use for prediction. Defaults to `1`. ~~int (option)~~                                                                                                                  |
| `--n-process`, `-n`                       | Number of processes to use for prediction. Defaults to `1`. ~~int (option)~~                                                                                                         |
```
